### ISSUE TITLE
if the applySSLSecurityPolicy:sender failed, the up layer will not be noticed

### ISSUE THREAD

**Issue by leimingdu** [novastone-media/MQTT-Client-Framework#150]
Tre is not error could be threw out.

```
if (eventCode &  NSStreamEventHasBytesAvailable) {
        DDLogVerbose(@"[MQTTCFSocketDecoder] NSStreamEventHasBytesAvailable");
        if (![self applySSLSecurityPolicy:sender withEvent:eventCode]){
            self.state = MQTTCFSocketDecoderStateError;
            self.error = [NSError errorWithDomain:@"MQTT"
                                             code:errSSLXCertChainInvalid
                                         userInfo:@{NSLocalizedDescriptionKey: @"Unable to apply security policy, the SSL connection is insecure!"}];
        }
    }

    [super stream:sender handleEvent:eventCode];
```

**Comment by jcavar** [comment]
@ckrey @leimingdu can you please explain a bit more in detail what is bug here exactly?

**Comment by jcavar** [comment]
Ok, we experienced this ourselves and I understand the issue now. So when pinning fails error property is updated but `MQTTSession` state itself is not.

**Comment by jcavar** [comment]
Here is PR for this:
https://github.com/novastone-media/MQTT-Client-Framework/pull/420/

I would appreciate your feedback if you have some time to review.

**Comment by jcavar** [comment]
This is fixed in 0.11.0
